precompiles/multi-asset-delegation/src/lib.rs

```diff
@@ -223,15 +242,26 @@ where
 		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
 
 		let caller = handle.context().caller;
-		let pallet_address = pallet_multi_asset_delegation::Pallet::<Runtime>::pallet_evm_account();
-		let who = Runtime::AddressMapping::into_account_id(caller);
 
-		let (deposit_asset, amount) = match (asset_id.as_u128(), token_address.0 .0) {
+		let (who, deposit_asset, amount) = match (asset_id.as_u128(), token_address.0 .0) {
 			(0, erc20_token) if erc20_token != [0; 20] => {
-				erc20_transfer(handle, token_address, pallet_address.into(), amount)?;
-				(Asset::Erc20(erc20_token.into()), amount)
+				let who = pallet_multi_asset_delegation::Pallet::<Runtime>::pallet_account();
+				let pallet_address =
+					pallet_multi_asset_delegation::Pallet::<Runtime>::pallet_evm_account();
+				erc20_transfer(
+					handle,
+					token_address,
+					caller.into(),
+					pallet_address.into(),
+					amount,
+				)?;
+				(who, Asset::Erc20(erc20_token.into()), amount)
 			},
-			(other_asset_id, _) => (Asset::Custom(other_asset_id.into()), amount),
+			(other_asset_id, _) => (
```

Comment:
Need tests for this branch please.

Reply:
covered here: https://github.com/tangle-network/tangle/blob/cd518abb9b89f88318cd144747d088b378270a78/precompiles/multi-asset-delegation/src/tests.rs#L104